editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,629 @@
+                study.clone().update_last_modified(conn).await?;
+
+                // Update project last_modification field
+                project.update_last_modified(conn).await?;
+
+                let scenario_with_details =
+                    ScenarioWithDetails::from_scenario(scenario, conn).await?;
+
+                let scenarios_response =
+                    ScenarioResponse::new(scenario_with_details, project, study);
+
+                Ok(scenarios_response)
+            }
+            .scope_boxed()
+        })
+        .await?;
+
+    Ok(Json(scenarios_response))
+}
+
+/// Return a specific scenario
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    responses(
+        (status = 200, body = ScenarioResponseV2, description = "The requested scenario"),
+        (status = 404, body = InternalError, description = "The requested scenario was not found"),
+    )
+)]
+#[get("")]
+async fn get(
+    db_pool: Data<DbPool>,
+    path: Path<ScenarioPathParam>,
+) -> Result<Json<ScenarioResponse>> {
+    use crate::modelsv2::Retrieve;
+
+    let ScenarioPathParam {
+        project_id,
+        study_id,
+        scenario_id,
+    } = path.into_inner();
+
+    let (project, study) = check_project_study(db_pool.clone(), project_id, study_id).await?;
+    let conn = &mut db_pool.get().await?;
+    // Return the scenarios
+    let scenario = Scenario::retrieve_or_fail(conn, scenario_id, || ScenarioError::NotFound {
+        scenario_id,
+    })
+    .await?;
+
+    // Check if the scenario belongs to the study
+    if scenario.study_id != study_id {
+        return Err(ScenarioError::NotFound { scenario_id }.into());
+    }
+
+    let scenarios_with_details = ScenarioWithDetails::from_scenario(scenario, conn).await?;
+    let scenarios_response = ScenarioResponse::new(scenarios_with_details, project, study);
+    Ok(Json(scenarios_response))
+}
+
+decl_paginated_response!(PaginatedResponseOfScenarioWithDetails, ScenarioWithDetails);
+
+/// Return a list of scenarios
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam, PaginationQueryParam, QueryParams),
+    responses(
+        (status = 200, body = PaginatedResponseOfScenarioWithDetails, description = "The list of scenarios"),
+    )
+)]
+#[get("")]
+async fn list(
+    db_pool: Data<DbPool>,
+    pagination_params: Query<PaginationQueryParam>,
+    path: Path<(i64, i64)>,
+    params: Query<QueryParams>,
+) -> Result<Json<PaginatedResponse<ScenarioWithDetails>>> {
+    let (page, per_page) = pagination_params
+        .validate(1000)?
+        .warn_page_size(100)
+        .unpack();
+    let (project_id, study_id) = path.into_inner();
+
+    let _ = check_project_study(db_pool.clone(), project_id, study_id).await?;
+    let ordering = params.ordering.clone();
+    let scenarios = Scenario::list(db_pool.clone(), page, per_page, (study_id, ordering)).await?;
```

Comment:
We might want to implement the trait `List` for `ScenarioWithDetails`. Currently, we're doing `n` psql queries which is never a good thing to do.